### ISSUE TITLE
Report says "Document does not have a meta description" when it actually does

### ISSUE THREAD

**Issue by mdorf85** [GoogleChrome/lighthouse#11659]
url: https://www.huffpost.com/entry/how-to-small-thanksgiving-dinner-covid_l_5f8de946c5b66ee9a5f2897b

lighthouse report:
![image](https://user-images.githubusercontent.com/7793502/98957328-0e333800-24cf-11eb-9ede-ac7f7fa3e87e.png)

page source:  (in the head of the document)
![image](https://user-images.githubusercontent.com/7793502/98957269-fcea2b80-24ce-11eb-9a3c-886da9da69f0.png)

lighthouse source code:
https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/audits/seo/meta-description.js

The check in the source code looks fine to me but unable to tell if it's really iterating over all meta elements just from this file.

**Comment by patrickhulce** [comment]
Thanks for filing @mdorf85! 

This one is really strange. We use `Element.prototype.matches` to determine if an element matches `head meta` and *only when requested by Lighthouse* does this return `false`. When loaded in DevTools the same function evaluation returns the description.

<details>
<summary>Code Snippets</summary>

The below snippet has `matches: true` outside of Lighthouse `matches: false` inside of Lighthouse.

```js
(() => {
      function getElementsInDocument(selector) {
  const realMatchesFn = window.__ElementMatches || window.Element.prototype.matches;
  /** @type {Array<HTMLElement>} */
  const results = [];

  /** @param {NodeListOf<HTMLElement>} nodes */
  const _findAllElements = nodes => {
    for (let i = 0, el; el = nodes[i]; ++i) {
      if (!selector || realMatchesFn.call(el, selector) || el.name === 'description') {
        results.push(el);
      }
      // If the element has a shadow root, dig deeper.
      if (el.shadowRoot) {
        _findAllElements(el.shadowRoot.querySelectorAll('*'));
      }
    }
  };
  _findAllElements(document.querySelectorAll('*'));

  return results;
};

      return getElementsInDocument('head meta').map(meta => {
        return {
          matches: meta.matches('head meta'),
          name: meta.name.toLowerCase(),
          content: meta.content,
          property: meta.attributes.property ? meta.attributes.property.value : undefined,
          httpEquiv: meta.httpEquiv ? meta.httpEquiv.toLowerCase() : undefined,
          charset: meta.attributes.charset ? meta.attributes.charset.value : undefined,
        };
      });
    })()
```

</details>

We'll need to investigate here.

**Comment by mdorf85** [comment]
Thank you for following up.

**Comment by adamraine** [comment]
@patrickhulce I am getting `matches: false` in devtools as well as lighthouse.

@mdorf85 On my end, it appears that the meta description is in `body` rather than `head`:
![Screen Shot 2020-11-18 at 5 44 44 PM](https://user-images.githubusercontent.com/6752989/99597266-c11afe80-29c5-11eb-8b97-532ba1f018c6.png)

**Comment by patrickhulce** [comment]
hm weird, I'm still not seeing that @adamraine 

![image](https://user-images.githubusercontent.com/2301202/99602793-b497a580-29c7-11eb-8276-8a0d76337990.png)

![image](https://user-images.githubusercontent.com/2301202/99602722-9b8ef480-29c7-11eb-90e9-146cb714cf03.png)

Whatever the difference is between your env and my env is probably whatever Lighthouse is seeing. I just enabled the standard Moto G4 from DevTools emulation and loaded the page.

**Comment by adamraine** [comment]
On windows I saw it in `head` consistently.
On linux and Mac I saw it in `body` consistently.

Using the Moto G4 in DT did not change the result for me.

**Comment by patrickhulce** [comment]
> On windows I saw it in head consistently.
On linux and Mac I saw it in body consistently.

The plot thickens, I've only tested this issue on macOS 10.15.5 so far. I wonder if geo or cookie settings could at all play a role? *tests theory* Huzzah! When tested in a fresh Chrome Canary profile the `meta[name=description]` appears in the `body` for me as well.

Seems like Lighthouse is in fact identifying a legitimate issue with this page on conditional serving @mdorf85 .
